data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-api/src/main/java/org/eclipse/edc/protocol/dsp/transferprocess/api/controller/DspTransferProcessApiController.java

```diff
@@ -176,17 +222,32 @@ public void transferProcessCompletion(@PathParam("id") String id, JsonObject jso
      * @param id         the ID of the process
      * @param jsonObject the {@link TransferTerminationMessage} in JSON-LD expanded form
      * @param token      the authorization header
+     * @return empty response or error.
      */
     @POST
     @Path("{id}" + TRANSFER_TERMINATION)
-    public void transferProcessTermination(@PathParam("id") String id, JsonObject jsonObject, @HeaderParam(AUTHORIZATION) String token) {
-        handleMessage(MessageSpec.Builder.newInstance(TransferTerminationMessage.class)
-                .processId(id)
-                .expectedMessageType(DSPACE_TRANSFER_TERMINATION_TYPE)
-                .message(jsonObject)
-                .token(token)
-                .serviceCall(protocolService::notifyTerminated)
-                .build());
+    public Response transferProcessTermination(@PathParam("id") String id, JsonObject jsonObject, @HeaderParam(AUTHORIZATION) String token) {
+        try {
+            handleMessage(MessageSpec.Builder.newInstance(TransferTerminationMessage.class)
```

Comment:
can't `handleMessage` just return a `Result` (or suitable subclass)? That way, we could keep in line with our coding style (returning results), and it would likely be much more efficient than throwing and immediately catching exceptions. 

Distinction between Auth failure and Invalid request could be handled through different `Failure` objects.

Reply:
Created #3017.